lib/cppcheck.cpp

```diff
@@ -562,10 +562,9 @@ unsigned int CppCheck::check(const FileWithDetails &file)
     return checkFile(file, emptyString);
 }
 
-unsigned int CppCheck::check(const FileWithDetails &file, const std::string &content)
+unsigned int CppCheck::check(const FileWithDetails &file, const uint8_t* data, std::size_t size)
```

Comment:
I feel that this interface is less safe.

Reply:
We could keep the `std::string` version for convenience. I got rid of it to make sure we always specify a buffer and because it was unused now.